docs/src/main/tut/docs/cassandra/README.md

```diff
@@ -4,6 +4,265 @@ title: Cassandra
+placeholders for real values - or a raw query string. Bear in mind that running a raw query using the 
+SessionAPI directly is unsafe and does not check query correctness at compile time. 
+
+```
+@free
+trait SessionAPI {
+
+  def init: FS[Session]
+
+  def close: FS[Unit]
+
+  def prepare(query: String): FS[PreparedStatement]
+
+  def prepareStatement(statement: RegularStatement): FS[PreparedStatement]
+
+  def execute(query: String): FS[ResultSet]
+
+  def executeWithValues(query: String, values: Any*): FS[ResultSet]
+
+  def executeWithMap(query: String, values: Map[String, AnyRef]): FS[ResultSet]
+
+  def executeStatement(statement: Statement): FS[ResultSet]
+
+  def executeWithByteBuffer(
+      query: String,
+      values: List[SerializableValueBy[Int]],
+      consistencyLevel: Option[ConsistencyLevel] = None): FS[ResultSet]
+
+}
+```
+
+### StatementAPI
+Provides methods to bind real query values to an already existing PreparedStatement, returning a 
+BoundStatement, which can now be ran in a safe way.
+
+```
+@free
+trait StatementAPI {
+
+  def bind(preparedStatement: PreparedStatement): FS[BoundStatement]
+
+  def setByteBufferByIndex(
+      boundStatement: BoundStatement,
+      index: Int,
+      bytes: ByteBuffer): FS[BoundStatement]
+
+  def setByteBufferByName(
+      boundStatement: BoundStatement,
+      name: String,
+      bytes: ByteBuffer): FS[BoundStatement]
+
+  def setValueByIndex[T](
+      boundStatement: BoundStatement,
+      index: Int,
+      value: T,
+      codec: ByteBufferCodec[T]): FS[BoundStatement]
+
+  def setValueByName[T](
+      boundStatement: BoundStatement,
+      name: String,
+      value: T,
+      codec: ByteBufferCodec[T]): FS[BoundStatement]
+
+  def setByteBufferListByIndex(
+      preparedStatement: PreparedStatement,
+      values: List[SerializableValueBy[Int]]): FS[BoundStatement]
+
+  def setByteBufferListByName(
+      preparedStatement: PreparedStatement,
+      values: List[SerializableValueBy[String]]): FS[BoundStatement]
+}
+```
+
+### ResultSetAPI
+Provides methods to interact directly with a Cassandra ResultSet, so we can get the final Object 
+representation, a list of them, or describe that that ResultSet could not have been found in the 
+current Cassandra instance.
+
+```
+@free
+trait ResultSetAPI {
+
+  def read[A](resultSet: ResultSet)(implicit FR: FromReader[A]): FS[A]
+
+  def readOption[A](resultSet: ResultSet)(implicit FR: FromReader[A]): FS[Option[A]]
+
+  def readList[A](resultSet: ResultSet)(implicit FR: FromReader[A]): FS[List[A]]
+
+}
+```
+
+## Freatures
+
+### Low level Queries
+The Low-Level API allows us to define queries with placeholders for query constraints, which can be 
+later bound to the final values, running the query in a safe mode, preventing CQL injection. 
+Frees-Cassandra checks the generated query is a valid one at compile time, throwing errors in case 
+any input value doesn't match with the expected data type. 
```

Comment:
I'm not sure about it, currently, we don't have type checking at compile time, am I'm missing something?

Reply:
Yep, hehe, you're right, I got excited too soon 😝